packages/react-button/src/components/MenuButton/useExpanded.tsx

```diff
@@ -68,17 +68,28 @@ export const useExpanded = <TDraftState extends ExpandedState>(draftState: TDraf
     [onKeyDown, setExpandedValue],
   );
 
-  // Assign extra props to the menu slot.
-  draftState.menu = {
-    target: rootRef,
-    ...draftState.menu,
-    onDismiss: React.useCallback(() => {
-      onMenuDismiss?.();
+  const onDismiss = React.useCallback(() => {
+    onMenuDismiss?.();
 
-      setExpandedValue(false);
+    setExpandedValue(false);
 
-      // TODO: should we re-focus the root?
-    }, [onMenuDismiss, setExpandedValue]),
+    // TODO: should we re-focus the root?
+  }, [onMenuDismiss, setExpandedValue]);
+
+  // Assign extra props to the menu slot.
+  draftState.menu = {
```

Comment:
Aren't you overwriting the menu object here?

Reply:
I'm only using the already passed `Menu` object (which is passed as `draftState.menu.children`) and putting it inside the `ExpandedContext.Provider`. So in essence the only thing I'm doing here is making the user able to use the `Context` as they see fit.

Reply:
Removing `Context` and passing `expand` as a prop.